taf/git.py

```diff
@@ -642,6 +643,26 @@ def clone(
         if self.default_branch is None:
             self.default_branch = self._determine_default_branch()
 
+    def clone_from_disk(
+        self, local_path: Path, remote_url: Optional[str] = None, is_bare: bool = False
+    ) -> None:
```

Comment:
I ran taf repo update on my local repositories and new local clone feels really slow (I didn't run a profiler). The problem is that we want updater performance to get faster 😞.

We should discuss/investigate if there are any approaches which we haven't considered here, since we want updater performance. Since the local disk clone is now slower, are symlinks disabled by default for git? I thought the local disk clone was supposed be instantaneous? 
Does this do a full copy of the directory contents? If so, I'd expect this to be a problem for law-html, since the repo size can get gigantic (>3 GB).  We'll also have a problem on the dev server, where the disk volume is 30GB, so I don't think we can reasonably expect to have full clones for 10 authentication repositories to work, right?

Reply:
We do want the performance to be improved, but this feature is aimed at addressing security concerns. We used to clone target repositories directly to their final location, just without checkout. Fetch also pulls down new changes, but will not merge them unless validation is successful. However, if it is not successful, these fetched chances will not get removed. This PR will first clone repositories to temp as bare repositories and validate them. If the validation is successful, they are cloned from that temp directory. These temp repositories are removed in any case. When running update, the repositories are clone from user's repository to temp, changes are fetched, and if the validation succeeds, user's repositories are updated by fetching from disk. So, we have new operations on top of everything else that we were doing, so this can, at best, be slightly slower than it was. This feature was not developed as a part of performance improvement efforts. We should figure out how much slower this is and if the performance can stay roughly the same as it was.